### PR TITLE
Social: Add SIG preview component

### PR DESCRIPTION
**Pull request by gmjuhasz**
This change introduces a new component that renders the current preview for SIG. It goes to the API endpoint that was introduced in #29596, which returns the token.

## Proposed changes:

* Created new component
  * The image URL is updated each time a needed property changes
  * String fields are debounced to 1500ms, to decrease calls.
* Added component tests for the new component

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1204116019981263-as-1204165892483477
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Checkout this PR, buy the Advanced Plan.
* Go to new post
* In the editor SIG panel, you should see the default generated preview
* Change the title, the title should change on the image after 1500ms.
* Set a featured image, it should show behind the title
* Set a custom message, that should overwrite the title on the image. While having a custom message, changes to the title should not cause any API calls.
* Pick a custom image, that should overwrite the featured image.

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/generated-image-preview/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * Fetches the preview of the generated image based on the post info
+ *
+ * @returns {React.ReactNode} The generated image preview.
+ */
+export default function GeneratedImagePreview() {
+	const [ generatedImageUrl, setGeneratedImageUrl ] = useState( null );
+	const [ isLoading, setIsLoading ] = useState( true );
+
+	const { customText, imageType, imageId } = useImageGeneratorConfig();
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/components/generated-image-preview/index.js]: Just looking at this, we're going to need the URL of the image, not the ID. I guess we're able to get that?

Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/generated-image-preview/index.js]: Yeah, I found it in Social Previews, so I used the same solution

File: projects/js-packages/publicize-components/src/components/generated-image-preview/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	const [ isLoading, setIsLoading ] = useState( true );
+
+	const { customText, imageType, imageId, template } = useImageGeneratorConfig();
+	const { getMedia } = select( 'core' );
+	const { title, featuredImage } = useSelect( _select => ( {
+		title: _select( editorStore ).getEditedPostAttribute( 'title' ),
+		featuredImage: _select( editorStore ).getEditedPostAttribute( 'featured_media' ),
+	} ) );
+
+	const imageTitle = useMemo( () => customText || title || 'Your Title', [ customText, title ] );
```
Comment (danielpost) [inline projects/js-packages/publicize-components/src/components/generated-image-preview/index.js]: Do we need `Your Title` here? Can it be an empty string? If not, should we make it translateable?

**Review by danielpost — DISMISSED** [review]
This works as expected, nice work!

File: projects/js-packages/publicize-components/src/components/generated-image-preview/index.js (hunk 2 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/generated-image-preview/index.js]: As discussed, removed this and added a header for the preview.
p1679930151849559/1679928768.658069-slack-C02JJ910CNL

**Review by danielpost — APPROVED** [review]

**Pull request merged**